packages/array-insert/index.tests.ts

```diff
@@ -0,0 +1,31 @@
+import insert from './index'
+
+// OK
+const arr1: number[] = [1, 2, 3, 4, 5, 6];
+const arr2: string[] = ['a', 'b', 'c'];
+let test1: (number | string)[]
+test1 = insert(arr1, arr2);
+test1 = insert(arr1, 'x');
+test1 = insert(arr1, arr2, 0);
+test1 = insert(arr1, arr2, arr1.length + 1);
+test1 = insert(arr1, 'x', 1);
+test1 = insert(arr1, arr2, undefined); // throw
```

Comment:
It throws an error but the third parameter should accept undefined type because it is optional.